Form-Builder/src/app/app.component.html

```diff
@@ -0,0 +1,71 @@
+<form class="form" [formGroup]="profileForm" (ngSubmit)="onSubmit()">
+
+  <div class="field">
+    <label class="label">username</label>
+    <input type="text" class="input" formControlName="username" 
+    [ngClass]= "{'invalid': profileForm.get('username')?.invalid && (profileForm.get('username')?.touched || profileForm.get('username')?.dirty)}"/>
```

Comment:
Create a function that checks all these conditions and returns only a boolean to simplify reading the html code